packages/agoric-cli/integration-tests/test-workflow.js

```diff
@@ -35,17 +35,32 @@ test('workflow', async t => {
     // Kill an entire process group.
     const pkill = (cp, signal = 'SIGINT') => process.kill(-cp.pid, signal);
 
+    function pspawnStdout(...args) {
+      let output = '';
+      const ps = pspawn(...args);
+      ps.cp.stdout.on('data', chunk => {
+        output += chunk.toString('utf-8');
+      });
+      ps.then(ret => {
+        if (ret !== 0) {
+          process.stdout.write(output);
+        }
+      });
+      return ps;
+    }
+
     // Run all main programs with the '--sdk' flag if we are in agoric-sdk.
     const extraArgs = fs.existsSync(`${__dirname}/../../cosmic-swingset`)
       ? ['--sdk']
       : [];
-    const myMain = args => {
+    function myMain(args) {
       // console.error('running agoric-cli', ...extraArgs, ...args);
-      return pspawn(`agoric`, [...extraArgs, ...args], {
+      return pspawnStdout(`agoric`, [...extraArgs, ...args], {
         stdio: ['ignore', 'pipe', 'inherit'],
```

Comment:
I would find it less surprising if instead of `pspawnStdout` above, just changing 'pipe' to 'inherit' here.

Reply:
Thanks for the comment.  I really only want to write stdout if the command failed.  Maybe that's unjustified.  I'll revisit this later.